### ISSUE TITLE
Refactor `U8TO32_LITTLE` in ChaCha20 code

### ISSUE THREAD

**Issue by briansmith** [briansmith/ring#133]
```
#define U8TO32_LITTLE(p)                              \
  (((uint32_t)((p)[0])) | ((uint32_t)((p)[1]) << 8) | \
   ((uint32_t)((p)[2]) << 16) | ((uint32_t)((p)[3]) << 24))
```

This should become `from_le_u32_ptr` in crypto/internal.h, alongside `from_be_u32_ptr` and friends. Also, we found that clang didn't optimize the generic code in `from_be_u32_ptr` well, so we should investigate whether we need to do the `memcpy` trick that we used for `from_be_u32_ptr`.

**Comment by briansmith** [comment]
crypto/poly1305/poly1305.c has:

```c
/* We can assume little-endian. */
static uint32_t U8TO32_LE(const uint8_t *m) {
  uint32_t r;
  memcpy(&r, m, sizeof(r));
  return r;
}
```

These should get merged together

**Comment by frewsxcv** [comment]
I'll take this.

**Comment by briansmith** [comment]
Fixed in #164 and #329. Then #351 and the similar commit for ChaCha20 removed the code altogether!

Thanks @frewsxcv and @PeterReid.
